nservicebus/messaging/discard-old-messages.md

```diff
@@ -29,3 +29,17 @@ To discard a message when a specific time interval has elapsed:
 
 <!-- import DiscardingOldMessagesWithCode -->
 
+## Caveats
+TimeToBeReceived relies on underlying functionality in the transport infrastructure to discard expired messages. This features usefulness is highly affected by the actual implementation in the different transports.
+
+#### MSMQ transport
+MSMQ continuously checks the TimeToBeReceived attribute of all queued messages. As soon as the message has expired, it is removed from the queue, and disk space reclaimed. Note that underlying issues in the MSMQ infrastructure makes it impossible for us to support setting TimeToBeReceived for messages sent from a transactional endpoint.
+
+#### RabbitMQ transport
+RabbitMQ also continuously checks the TimeToBeReceived attribute, but only for the top message on each queue. Expired messages are not removed from the queue, and their disk space is not reclaimed until they reach the front of the queue. Using TimeToBeReceived as a disk saving measure on RabbitMQ is not a great choice for queues with long-lived messages like audit and forward. The TimeToBeReceived of other messages in front of a message in a queue will affect when the message is actually removed.
```

Comment:
Hmm if I am overriding TTBR for audits and doing it consistently for each message, does it mean that the disk space will be reclaimed? Or do I actually need to actively receive a message in the front for rabbit to check the TTBR and delete that message?